parser/openfoodfacts_taxonomy_parser/parser.py

```diff
@@ -0,0 +1,283 @@
+        if lc in self.stopwords :
+            words_to_remove = self.stopwords[lc]
+            new_words = []
+            for word in words.split("-"):
+                if word not in words_to_remove :
+                    new_words.append(word)
+            return ("-").join(new_words)
+        else : return words
+
+    def add_line(self,line):
+        """to get a normalized string but keeping the language code "lc:" and the "," (commas) separators , used to add an id to entries and to add a parent tag"""
+        if len(line)<=3:return self.normalizing(line)
+        else:
+            lang=line[:2]
+            new_line=lang+":"
+            for word in line[3:].split(","):
+                new_line+=self.remove_stopwords( lang,self.normalizing(word,lang) ) +','
+            new_line=new_line[:-1]
+            return new_line
+
+    def get_lc_value(self,line):
+        """to get the language code "lc" and a list of normalized values"""
+        lc=line[:2]
+        new_line=[]
+        for word in line[3:].split(","):
+            new_line.append( self.remove_stopwords( lc,self.normalizing(word,lc) ) )
+        return lc,new_line
+
+    def new_node_data(self):
+        """ To create an empty dictionary that will be used to create node """
+        data = {
+                "id" : '',
+                "main_language" : '',
+                "comment" : [],
+                "parent_tag" : [],
+                'src_position' : 0,
+        }
+        return data
+
+    def set_data_id(self,data,id,line_number):
+        if not data['id'] :
+            data['id']=id
+        else :
+            raise DuplicateIDError(line_number)
+        return data
+
+    def header_harvest(self,filename):
+        """ to harvest the header (comment with #), it has its own function because some header has multiple blocks """
+        h=0
+        header=[]
+        for _,line in self.file_iter(filename):
+            if not(line) or line[0]=='#':
+                header.append(line)
+            else : break
+            h+=1
+
+        # we don't want to eat the comments of the next block and it remove the last separating line
+        for i in range(len(header)):
+            if header.pop():
+                h-=1
+            else : break
+
+        return header,h
+
+    def harvest(self,filename):
+        """Transform data from file to dictionary"""
+        index_stopwords=0
+        index_synonyms=0
+        language_code_prefix = re.compile('[a-zA-Z][a-zA-Z]:')
+        self.stopwords=dict()
+
+        #header
+        header,next_line = self.header_harvest(filename)
+        yield header
+        self.is_before = '__header__'
+
+        #the other entries
+        data = self.new_node_data()
+        for line_number,line in self.file_iter(filename,next_line):
+            if line == '': # can be the end of an block or just 2 line separators, file_iter() always end with ''
```

Comment:
@BryanH01 Do try to shorten the comments, since it's a little too long to read. As a  convention, do try to restrict shorter comments to the side and longer comments above a line.

Reply:
```suggestion
            if line == '': 
                # can be the end of an block or just 2 line separators, 
                # file_iter() always end with ''
```